File: programmers/LV1/약수의 개수와 덧셈.js
// 약수의 개수와 덧셈

function solution(left, right) {
  let answer = 0;
  let count = 0;

  for (let i = left; i <= right; i++) {
    count = 0;
    for (let j = 1; j <= i; j++) {
      if (i % j == 0) {
        count++;
      }
    }

    if (count % 2 === 0) {
      answer += i;
    } else {
      answer -= i;
    }
  }

  return answer;
}

function solution(left, right) {
  var answer = 0;
  for (let i = left; i <= right; i++) {
    if (Number.isInteger(Math.sqrt(i))) {
      // 제곱근이 정수면 약수의 개수는 홀수
      answer -= i;
    } else {
      answer += i;
    }
  }
  return answer;
}

File: programmers/python/약수의 개수와 덧셈.py
# 약수의 개수와 덧셈
def solution(left, right):
    answer = 0

    for i in range(left, right+1):
        cnt = 1
        num = 1
        while num < i:
            if i%num == 0:
                cnt += 1
            num += 1

        if cnt % 2 == 0:
            answer += i
        else:
            answer -= i

    return answer

# 제곱수는 약수의 개수가 홀수
def solution2(left, right):
    answer = 0
    for i in range(left,right+1):
        if int(i**0.5)==i**0.5:
            answer -= i
        else:
            answer += i
    return answer